service/tbc/tbc.go

```diff
@@ -87,17 +88,26 @@ type Config struct {
 	PrometheusListenAddress string
 	PprofListenAddress      string
 	Seeds                   []string
+
+	// Fields used for running TBC in External Header Mode, where P2P is disabled
+	// and TBC is used to determine consensus based on headers fed from external
+	// code that manages the TBC node.
+	ExternalHeaderMode      bool              // Whether Header-Only Mode is enabled
+	EffectiveGenesisBlock   *wire.BlockHeader // The header to use as the first block in TBC's consensus view
+	GenesisHeightOffset     uint64            // The height of the effective genesis block
+	GenesisDifficultyOffset big.Int           // The cumulative difficulty of the effective genesis block
```

Comment:
This is theoretical, but should we create a fake blockchain params here? so that we don't have to if/then/else all over the place.